### ISSUE TITLE
5.4.1 not building in homebrew

### ISSUE THREAD

**Issue by ryansch** [minetest/minetest#11233]
##### Minetest version

```
5.4.1
```

##### OS / Hardware

Operating system: macOS

##### Summary

https://github.com/Homebrew/homebrew-core/runs/2405681581?check_suite_focus=true
There are multiple closed PRs trying to update brew's minetest version to 5.4.1 but the build fails on macOS 10.14. I wanted to draw attention to it here.

**Comment by sfan5** [comment]
5.4.1 has had no changes in this area so that shouldn't be it.
Looking at the error there's rather some issue with Irrlicht itself:
<pre>
Undefined symbols for architecture x86_64:
  "_objc_opt_self", referenced from:
      irr::CIrrDeviceMacOSX::minimizeWindow() in libIrrlicht.a(CIrrDeviceMacOSX.o)
      irr::CIrrDeviceMacOSX::restoreWindow() in libIrrlicht.a(CIrrDeviceMacOSX.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/minetest] Error 1
</pre>

Googling a bit it looks like this can indicate a mismatch in macOS APIs used (e.g. [here](https://github.com/marta-file-manager/marta-issues/issues/729)).

**Comment by ryansch** [comment]
Does 5.4.1 build with Irrlicht or IrrlichtMt?

**Comment by ryansch** [comment]
I also took a stab at creating a formula for IrrlichtMt. I have no idea if that's the right way forward or if there's a way to build it as part of the existing minetest formula.

**Comment by sfan5** [comment]
5.4.1 builds with Irrlicht only.

**Comment by ryansch** [comment]
Thanks!

**Comment by Jordach** [comment]
> 5.4.1 has had no changes in this area so that shouldn't be it.
> Looking at the error there's rather some issue with Irrlicht itself:
> 
> Undefined symbols for architecture x86_64:
>   "_objc_opt_self", referenced from:
>       irr::CIrrDeviceMacOSX::minimizeWindow() in libIrrlicht.a(CIrrDeviceMacOSX.o)
>       irr::CIrrDeviceMacOSX::restoreWindow() in libIrrlicht.a(CIrrDeviceMacOSX.o)
> ld: symbol(s) not found for architecture x86_64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> make[2]: *** [bin/minetest] Error 1
> Googling a bit it looks like this can indicate a mismatch in macOS APIs used (e.g. [here](https://github.com/marta-file-manager/marta-issues/issues/729)).

This issue likes to appear when making Universal Binaries containing x64 and arm64. I get the same on M1 Macs irregardless.

**Comment by gvoc** [comment]
Same here, that is because Minetest can't build on ARM

**Comment by animaux** [comment]
> Same here, that is because Minetest can't build on ARM

Tried to compile on an M1 on Monterey via readme instructions, process gets stuck at `make -j$(nproc)`. No progress after:

```sh
[ 99%] Building CXX object src/CMakeFiles/minetest.dir/voxelalgorithms.cpp.o
```

Is there a chance to have a native M1 version anytime?

**Comment by SmallJoker** [comment]
@animaux

> process gets stuck at make -j$(nproc). No progress after:

CMake generates a regular Makefile build script. There should be no difference to compiling any other application. It seems that you did everything right. Either the compiler or the linker would throw an error if something goes wrong in that situation. I don't think this is related to Minetest specifically and might be caused by a slow linker (i.e. compiler performance issues).

**Comment by animaux** [comment]
@SmallJoker thanks! 

So you think I should let it run for a while and it should eventually finish? All the other items went through quite fast, but when arriving at the mentioned item it seems like it is stuck in a loop (?), with persistent very high CPU usage.

Anything I else could do to help debug this?

**Comment by sfan5** [comment]
We just released 5.5.0 which changed up a lot related stuff and to my knowledge it (at least) builds fine on macOS.
If the same/similar problem also affects the new release please open a new issue.
